### ISSUE TITLE
Certain groups very slow to update.... that used to work fine.

### ISSUE THREAD

**Issue by stevenmcastano** [nZEDb/nZEDb#2370]
###Expected behaviour
My normal tmux scripts tool about 25 - 30 minutes to update an entire round of binary updates.

###Actual behaviour
Now the same scritpts that are running take just over an hour to 90 minutes and seem to get stuck on 3 groups, teevee, moovee and inner-sanctum.

so I'm trying to run updates on those manually since they're falling behind and holding everything else up, but when I do I'm getting the following output:

```
Updating group: alt.binaries.moovee
Processing alt.binaries.moovee

Part repair enabled. Checking for missing parts.

Group alt.binaries.moovee has 36,104,482 new articles. Leaving 354,482 for next pass.
Server oldest: 2 Server newest: 462,408,729 Local newest: 426,304,247


Getting 250,000 articles (426,304,248 to 426,554,247) from alt.binaries.moovee - (35,500,000 articles in queue).
Received 21242KB from group (alt.binaries.moovee).
```

And it just seems to be hanging there for all 3. It appears like it fetch the data from the newgroup server, but maybe it's hanging on trying to put it into the DB? Should I reset those tables? Maybe they got corrupt? Delete them, re-add and backfull about 30,000,000 articles... 

what might be causing this? And what's the best way to fix it?

**Comment by DariusIII** [comment]
If you have 30+M articles to update in moovee, you are doing something wrong. And to be honest, the whole binaries->releases->postprocess cycle in normal run with tmux takes couple of minutes, if you run it properly. My scripts all run with 30 seconds between runs. You have set values that are too long for your server to handle properly.

**Comment by stevenmcastano** [comment]
They were never that way, I'm just running them manually to catch up in the background. My normal updates only grab 1 million max and most groups have way less than that, but even when I was running 5 million per session, it would still grab chunks of 250,000 pretty damn quickly.... and does for all the other groups.

Just those 3 have a problem. I've been running with the same setting for a year and there's never been an issue, then all of a sudden about two weeks ago this popped up.

How does the reset, purge and delete options work? I'm thinking maybe the tables for them got hosed or something and a restart would help.

**Comment by Wally73** [comment]
prob a spam poster also busy so block that one

**Comment by stevenmcastano** [comment]
A spam poster? What do you mean block it?

**Comment by stevenmcastano** [comment]
The thing is, it's requesting the articles from the server and getting them pretty quickly. It seems like the slowdown is in putting them into the actual database. Could that be possible?

**Comment by stevenmcastano** [comment]
As I do more digging, it looks like I have 4 groups that have stopped updating as of about 7 days ago. They are groups with a ton of releases in them.

Is there a way to totally reset the database for those 4 groups, but keep all the current releases, then start updating them again, and backfill back to the 7 or 8 days I'm missing?

**Comment by stevenmcastano** [comment]
Now, just to try it... I reset the database for one of the broken groups, and tried manually updating the binaries again and it seem seems to be taken a very long time. I get the message that it's grabbed 100,000 articles from the server very quickly, but then it just sits there and hangs and doesn't appear to every be sending them to the database properly.

any idea on what I can do next?

**Comment by stevenmcastano** [comment]
I finally got SOMETHING out of it:

```2.69s to download articles, 699.62s to process collections, 4.71s to insert binaries/parts, 0.00s for part repair, 707.01s total.```

Why would it take so long to process collections? Is something wrong?

**Comment by stevenmcastano** [comment]
Another REALLY long collections time, with some PHP error:

```
Updating group: alt.binaries.teevee
Processing alt.binaries.teevee

Part repair enabled. Checking for missing parts.

Group alt.binaries.teevee has 30,869,016 new articles. Leaving 369,016 for next pass.
Server oldest: 2 Server newest: 1,225,617,711 Local newest: 1,194,748,695


Getting 250,000 articles (1,194,748,696 to 1,194,998,695) from alt.binaries.teevee - (750,000 articles in queue).
Received 23006KB from group (alt.binaries.teevee).
Debug: Sending 21571.54 KB of parts to MySQL
Received 250000 articles of 250,000 requested, 0 blacklisted, 1 not yEnc.
7.22s to download articles, 3,296.31s to process collections, 15.21s to insert binaries/parts, 0.00s for part repair, 3,318.74s total.

Getting 250,000 articles (1,194,998,696 to 1,195,248,695) from alt.binaries.teevee - (500,000 articles in queue).
PHP Strict Standards:  Only variables should be assigned by reference in /usr/share/php/PEAR.php on line 601
PHP Strict Standards:  Only variables should be assigned by reference in /usr/share/php/PEAR.php on line 601
Received 22805KB from group (alt.binaries.teevee).
```

**Comment by DariusIII** [comment]
These errors are not nZEDb related, they are from PEAR.php which is part of PHP and we have no control of. Can you add a screenshot of your tmux monitor window (all the panes from first window, where montor , update_releases, update_binaries are)?

**Comment by stevenmcastano** [comment]
![screen shot 2017-01-28 at 10 24 03 am](https://cloud.githubusercontent.com/assets/6171444/22397630/fd0c76be-e543-11e6-9ec5-e00d7a1b7667.png)

Is that what you're looking for?

**Comment by Bart39** [comment]
Is your front end separate from your db ? 2gb ram used seems awfully low for 33 mil releases in your db, also why is your swap usage 1.5gb when you still have nearly 10gb free. Maybe not related just those stats look odd

**Comment by DariusIII** [comment]
You definitely have a misconfigured server and database. High swap usage, low RAM and CPU usage, and 2M collections in line to be created. I would need to see your my.cnf. Do you use MyISAM or InoDB tables?

**Comment by stevenmcastano** [comment]
Database is a totally separate box with 24 cores, 128G of RAM the entire DB running on a RAM drive with plenty of RAM to spare and no swap being used, and at least a 40G InnoDB buffer, all tables are InnoDB.

Here's the conf: [http://pastebin.com/bZEYSjzJ](http://pastebin.com/bZEYSjzJ)

**Comment by niel** [comment]
You're also running on an older version, which will obviously be missing any fixes to known problems. That makes it redundant to ask about problems with nZEDb. Please update to v0.6.8.x

**Comment by stevenmcastano** [comment]
Running an older version of what? nZEDb? Should I do an upgrade then?

**Comment by DariusIII** [comment]
You will need to redo your database config, innodb_file_per_table needs to be 

> innodb_file_per_table = 1

Just setting a innodb_buffer_pool_size is not enough, there are certain tweaks that need to be done. 

You can start from here : [InnoDB performance optimization basics (redux)](https://www.percona.com/blog/2013/09/20/innodb-performance-optimization-basics-updated/)

And you did not answer what table types you use, MyISAM or InnoDB.

**Comment by stevenmcastano** [comment]
Everything is InnoDB, and sure I can add the innodb_file_per_table line.... but can that be done on the fly? Everything has been running fine for well over a year, if not longer without that.

**Comment by stevenmcastano** [comment]
It appears like this might have nothing to do with nZEDb or the database... I just a brand new install on Unbuntu 16.04, with the same DB server (MariaDB 10.1), and it seems to be doing the same thing. I'm only trying to index a single group with all the default settings, just a.b.teevee.... and already it's connecting to the newgroup server, grabbing articles ASAP, then hanging while trying to put things in the database...

My guess is it's still going to remove crazy long times dealing with collections.

Is nobody else seeing this happening?

**Comment by stevenmcastano** [comment]
Yup, 290 seconds to process collections on a 20,000 article grab.

**Comment by kaibosh** [comment]
If you haven't already a limit, try setting "Minimum Files To Form Release" >=2 on those groups. Have a look at the most frequent poster, consider blacklisting obvious spam.

Does mysqltuner give any recommendations?

**Comment by stevenmcastano** [comment]
mysqltuner says I'm all good bascially, yeah. Really no warnings at all.

I did just add the minimum files setting to those groups, and I think I added a regex to block some of what I saw coming in... we'll see if that helps.

**Comment by b3rs3rk** [comment]
The DB isn't his problem.  If you look at the timers, the issue is with processing the headers not inserting them .  Have you added any custom collections/blacklist regexes?

**Comment by stevenmcastano** [comment]
I haven't added any custom regex, but I did find some posts on the forum about a spammer who's been going a little crazy. Added a regex to block ".*pr3d@NET\.world.*" show's now that regex is being hit and the groups I was having a problem with are now showing at in some cases, on a 250,000 article fetch, upwards of 200,000 of them are blacklisted.

**Comment by stevenmcastano** [comment]
As @b3rs3rk mentioned... spam seems to be what's going this, I've now added two regexs to combat the major spammers:
```
.*pr3d@NET\.world.*
\[.*\] \\.*\\::.*\..*\..*\::\/.*\/ \[.*\] - yEnc.*
```
Those are taking care of a lot of it. In 3 or 4 groups now I'm grabbing 100,000 parts at a time, with around 80,000 to 90,000 parts blacklisted.

**Comment by DariusIII** [comment]
Yeah, i must apologize, as this morning i had the same thing happen to me, 8M collections waiting in line. Added a regex to battle that spam

**Comment by stevenmcastano** [comment]
@DariusIII Not a problem man, no apology needed... it took me a while to track this down as well. It wasn't until I noticed the collection size growing in certain tables and deadlock issues that I start pulling it apart and looking at what was in these huge tables.

Looks like most of the spamming is happening for me in:
a.b.teevee
a.b.moovee
a.d.inner-sanctum

And now with the 2 regexs I mentioned above in place, I'm back to indexing 283 groups in a cycle that's about 25 minutes, and that's while playing catchup on some of the groups that fell a few million articles behind.

**Comment by ThePeePs** [comment]
I also added ~~([0-9a-f]){9}@([0-9a-f]){9}\.([0-9a-f]){9}~~ ([0-9a-f]{9})@\1\.\1 as a poster regex.

**Comment by ghost** [comment]
I'm not seeing anything different to be honest but my site seems dead slow. To be completely fair I have 150+ms of latency on this wifi connection abroad...

I restarted tmux and it reset 2 mill collections.

Anything I need to change/add?
